File: frontend/src/api/memos.ts
import { apiClient } from './client'
import type { Memo, MemoCreate, MemoUpdate } from './generated'

/** 管理端个人备忘录：纯文本记录，一级 tab 快捷切换多条（按登录用户隔离）。 */
export const memoApi = {
  /** 当前用户的全部备忘录，最近更新的排前面。 */
  listMemos: () => apiClient.get<Memo[]>('/memos').then((r) => r.data),
  /** 新建一条备忘录（title 为空时后端回退为「未命名备忘录」）。 */
  createMemo: (payload: MemoCreate = { title: '', content: '' }) =>
    apiClient.post<Memo>('/memos', payload).then((r) => r.data),
  /** 更新标题 / 内容（乐观锁：version 必须与当前一致）。 */
  updateMemo: (id: number, payload: MemoUpdate) =>
    apiClient.patch<Memo>(`/memos/${id}`, payload).then((r) => r.data),
  /** 删除一条备忘录。 */
  deleteMemo: (id: number) => apiClient.delete(`/memos/${id}`).then((r) => r.data),
}

File: backend/app/api/v1/memos.py
"""管理端个人备忘录接口：纯文本，一级 tab 快捷切换多条。

所有登录用户（含只读角色）均可使用：个人备忘录按创建人隔离，
接口只操作当前用户自己的数据。错误遵循 docs/api-error-conventions.md
（资源不存在 → 400 + NOT_FOUND，版本冲突 → 409 + VERSION_CONFLICT）。
"""

from __future__ import annotations

from fastapi import APIRouter

from app.core.permissions import CurrentUser, DbSession
from app.models import Memo
from app.schemas import MemoCreate, MemoRead, MemoUpdate
from app.services import memo_service

router = APIRouter(prefix="/memos", tags=["备忘录"])


@router.get("", response_model=list[MemoRead])
async def list_memos(session: DbSession, user: CurrentUser) -> list[Memo]:
    return await memo_service.list_memos(session, user.id)


@router.post("", response_model=MemoRead, status_code=201)
async def create_memo(
    data: MemoCreate,
    session: DbSession,
    user: CurrentUser,
) -> Memo:
    return await memo_service.create_memo(session, user.id, data)


@router.patch("/{memo_id}", response_model=MemoRead)
async def update_memo(
    memo_id: int,
    data: MemoUpdate,
    session: DbSession,
    user: CurrentUser,
) -> Memo:
    return await memo_service.update_memo(session, memo_id, user.id, data)


@router.delete("/{memo_id}", status_code=204)
async def delete_memo(memo_id: int, session: DbSession, user: CurrentUser) -> None:
    await memo_service.delete_memo(session, memo_id, user.id)